### PR TITLE
add contributing documentation

### PR DESCRIPTION
**Pull request by mkorbi**
Signed-off-by: Max Körbächer <max@koerbaecher.io>

Add short documentation on how to get started with contributing to the WG.

### CODE REVIEW

File: CODE-OF-CONDUCT.md
Comment (rootfs) [inline CODE-OF-CONDUCT.md:48]: `tag-security`?

File: CODE-OF-CONDUCT.md
Comment (rootfs) [inline CODE-OF-CONDUCT.md:35]: `security`?

File: CODE-OF-CONDUCT.md
Comment (rootfs) [inline CODE-OF-CONDUCT.md:55]: `tag security`?

File: CODE-OF-CONDUCT.md
Comment (rootfs) [inline CODE-OF-CONDUCT.md:60]: Security Tag?

File: CODE-OF-CONDUCT.md
Comment (rootfs) [inline CODE-OF-CONDUCT.md:104]: security?

File: CODE-OF-CONDUCT.md
Comment (rootfs) [inline CODE-OF-CONDUCT.md:115]: Please also add [tag security code of conduct](https://github.com/cncf/tag-security/blob/main/CODE-OF-CONDUCT.md)  from which this draft is based on.

**Comment by mkorbi** [comment]
We don't have yet the mail list for leads

File: CODE-OF-CONDUCT.md
Comment (catblade) [inline CODE-OF-CONDUCT.md:118]: Another tag security.

**Comment by mkorbi** [comment]
Added the point from our last meeting that we want to utilize GH discussions to move discussions out of the mailing list/slack

File: CONTRIBUTING/first-time-contributions.md
Comment (catblade) [inline CONTRIBUTING/first-time-contributions.md]: Why are you removing the member list?

File: CONTRIBUTING/README.md
Comment (rootfs) [reply CONTRIBUTING/README.md:40]: +1

File: CONTRIBUTING/first-time-contributions.md
Comment (mkorbi) [reply CONTRIBUTING/first-time-contributions.md]: We didn't define when is someone a member, yet it doesn't have any value. Is everyone a member who is at the meeting? Or who PRed something? Or wrote something to the mailing list? 

All unclear, therefore removed.

File: CODE-OF-CONDUCT.md
Comment (catblade) [inline CODE-OF-CONDUCT.md:1]: Rather than write our own, can we follow what sig-security is and just follow our governing body's COC?  Here is the link to the CNCF one:  https://github.com/cncf/foundation/blob/main/code-of-conduct.md

File: CONTRIBUTING/first-time-contributions.md
Comment (catblade) [inline CONTRIBUTING/first-time-contributions.md:3]: Formatting here is weird.

File: CODE-OF-CONDUCT.md
Comment (leonardpahlke) [inline CODE-OF-CONDUCT.md:3]: Small nit  

```suggestion
We follow the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/main/code-of-conduct.md)
```

File: CODE-OF-CONDUCT.md
Comment (leonardpahlke) [inline CODE-OF-CONDUCT.md:46]: Small nit  

```suggestion
more details refer to the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/main/code-of-conduct.md)). If it is
```

File: CODE-OF-CONDUCT.md
Comment (leonardpahlke) [reply CODE-OF-CONDUCT.md:49]: Hmm, yes – I can understand that. Perhaps it would be good to explicitly state that you can also turn to someone you know / trust / are close to in order to express concerns?

File: CONTRIBUTING/README.md
Comment (leonardpahlke) [reply CONTRIBUTING/README.md:21]: IMO, I think it's ok to route to the README.md meeting section, which lists all the information.  

@catblade in the future in this case, could you write a suggestion that specifically outlines what should be added to the PR? (@catblade just a friendly comment, I hope you don't mind :slightly_smiling_face:)

File: CONTRIBUTING/first-time-contributions.md
Comment (leonardpahlke) [inline CONTRIBUTING/first-time-contributions.md:11]:
```suggestion
We happily welcome our new contributors to this community. 
If you are contributing to the CNCF and/or Environmental Sustainability WG for the first time it is okay if you feel overwhelmed. 
We, as a community, are always there to help you with any problems you are facing.
Open source is about collaboration and we are always there to support each other.
```

File: CODE-OF-CONDUCT.md
Comment (catblade) [reply CODE-OF-CONDUCT.md:49]: The CNCF has taken a lot of time and trouble writing their COC, so I think it most appropriate we follow what they do already.  They've already spent the time thinking of all the corner cases (or running into them painfully).

Comment (catblade) [reply CODE-OF-CONDUCT.md:49]: I don't think that helps either.  What if you are new to the community?  

Here is what the CNCF has on reporting.
https://github.com/cncf/foundation/blob/main/code-of-conduct.md#reporting

File: CODE-OF-CONDUCT.md
Comment (rootfs) [inline CODE-OF-CONDUCT.md:3]: Why re-invent yet another CoC? The CNCF CoC is designed for general use in all CNCF community. If the CNCF CoC is not able to serve the purpose here, please state the requirement and use case and bring the discussion to the TOC. It is way beyond the scope of this WG to re-scope the CNCF CoC.

File: CODE-OF-CONDUCT.md
Comment (rootfs) [inline CODE-OF-CONDUCT.md:36]: Again, this is a dup of the CNCF CoC. There is a reporting/enforcement process in place and I don't see why it cannot be used to handle the situation here.

File: CODE-OF-CONDUCT.md
Comment (leonardpahlke) [reply CODE-OF-CONDUCT.md:3]: > Why re-invent yet another CoC? 

Not really. It looks like this PR introduced the TAG Security CoC to the working group. 

IMO: let's move on quickly.... 
Option1: either merge this PR, which is fine from my point of view since we are reusing the TAG-Security structure, or. 
Option2: remove this file and refer to the CNCF CoC in a single line only.

I currently prefer option2. (I don't think the additional CoC stuff is necessary and could be confusing).

File: CODE-OF-CONDUCT.md
Comment (mkorbi) [reply CODE-OF-CONDUCT.md:1]: First sentence: We follow the [CNCF Code of Conduct][cncf-coc].
How much more we should use the CNCF CoC?

The first part explains how we want to follow this commitment. The second part clarifies incident handling which the CoC doesn't because every project has to manage this by itself.

The next part is about references, which are not managed in CoC.

File: CODE-OF-CONDUCT.md
Comment (rootfs) [reply CODE-OF-CONDUCT.md:3]: +1 
Let's leverage on the CNCF tooling and Keep the draft lean and fast forward to action.

File: CODE-OF-CONDUCT.md
Comment (mkorbi) [reply CODE-OF-CONDUCT.md:49]: It doesn't force anyone, it encourages people to speak up if they see something they don't like. What should be normal and is in no way outstanding. 

The CNCF CoC addresses this by going to a top-level escalation, rather than just simply solving this between people.

File: CONTRIBUTING/README.md
Comment (mkorbi) [reply CONTRIBUTING/README.md:21]: +1 on leo

Comment (catblade) [reply CONTRIBUTING/README.md:21]: Can we remove this section entirely, if you just want to route to the other README?

File: CONTRIBUTING/first-time-contributions.md
Comment (catblade) [inline CONTRIBUTING/first-time-contributions.md:42]: Instead of "We got you", let's replace "We are here to help you."

File: CODE-OF-CONDUCT.md
Comment (mkorbi) [reply CODE-OF-CONDUCT.md:3]: This review was already given, therefore it would be great to have this comment on the corresponding feedback.
However, we don't reinvent the CoC. As the first line states: We follow the CNCF CoC. 
How much more we should use the CNCF CoC?

The first part explains how we want to follow this commitment. The second part clarifies incident handling which the CoC doesn't because every project has to manage this by itself. (The reporting part of the CoC is like this - someone uses for example a religious emoji, because it looks like K8s, in the wrong context and rather as an adult person just solve this but highlight your concerns, you go to the highest escalation point known. Think this approach is not sustainable, and therefore mentioned encouraging people to bring up their points like you would in other cases.)

The next part is about references, which are not managed in CoC.

If this is such a problem we can move it somewhere else into own files or whatsoever.

File: CONTRIBUTING/README.md
Comment (mkorbi) [reply CONTRIBUTING/README.md:21]: No, to motivate communication should be present and highlighted as much as needed.

File: CONTRIBUTING/first-time-contributions.md
Comment (catblade) [reply CONTRIBUTING/first-time-contributions.md]: We really should consider a place where people can themselves as members.  It was really a welcoming feeling with sig-security when I was able to do so.

**Comment by catblade** [comment]
> lgtm

Should not be a lgtm until we have the community discussing something as serious as a code of conduct.

File: CODE-OF-CONDUCT.md
Comment (rootfs) [reply CODE-OF-CONDUCT.md:3]: > This review was already given, therefore it would be great to have this comment on the corresponding feedback. However, we don't reinvent the CoC. As the first line states: We follow the CNCF CoC. How much more we should use the CNCF CoC?
> 
If CNCF CoC is not able to cover the use case, voice the requirement and get the feedback before re-scoping it.

> The first part explains how we want to follow this commitment. The second part clarifies incident handling which the CoC doesn't because every project has to manage this by itself. (The reporting part of the CoC is like this - someone uses for example a religious emoji, because it looks like K8s, in the wrong context and rather as an adult person just solve this but highlight your concerns, you go to the highest escalation point known. Think this approach is not sustainable, and therefore mentioned encouraging people to bring up their points like you would in other cases.)
> 
Per CNCF CoC, there are two layers of escalation: project maintainer and CNCF. Any evidence why this process is not sustainable?
> The next part is about references, which are not managed in CoC.
Again, if this is not something covered by CNCF CoC, open an issue and have it resolved, it is beneficial to the community.
> 
> If this is such a problem we can move it somewhere else into own files or whatsoever.

**Comment by rootfs** [comment]
> lgtm

I don't want to disagree with the approval, but since there are larger issues here, more reviews are requested.

**Comment by leonardpahlke** [comment]
> Should not be a lgtm until we have the community discussing something as serious as a code of conduct.

Sure, we can wait until the next meeting to further discuss this.

> [...] but since there are larger issues here, more reviews are requested.

Sounds reasonable to me +1.

(Note about the `lgtm`: To me, the changes proposed here have been sufficiently discussed, and I am ok with approving the requested changes)

File: CONTRIBUTING/README.md
Comment (leonardpahlke) [inline CONTRIBUTING/README.md:44]:
```suggestion
If you need additional access rights for the TAG GitHub Repository and/or the [Project Board](https://github.com/orgs/cncf/projects/10), please open a [simple issue](https://github.com/cncf/tag-env-sustainability/issues/new) with the request.
```

I will add a note to the agenda to check the project board at the end of a meeting (if there is time) right now the project board is not very well maintained tbh.

**Review by leonardpahlke — CHANGES_REQUESTED** [review]
So, reading threw some of the previous comments ... 
* I would propose to remove the `CODE-OF-CONDUCT.md` content and add just a reference to the cncf code of conduct.
* We also need to replace `WG` references with `TAG`.
* And the line breaks are not formatted correct.

**Comment by mkorbi** [comment]
For now I took out the CoC stuff, I still believe we can do better, but that discussion is moved to the CoCC. 
Adjusted wording and WG/TAG things.

File: README.md
Comment (leonardpahlke) [inline README.md:30]: shorten this?

```suggestion
## New contributors

We encourage new contributors to read the [contributing documentation](CONTRIBUTING.md). If you have any questions, please contact us through the channels mentioned above.
```

**Review by leonardpahlke — CHANGES_REQUESTED** [review]
How about shortening this to make it more concise? In the end, the way to participate in this TAG is very similar to the other TAGs and other open source projects in the CNCF.

**Review by leonardpahlke — APPROVED** [review]
LGTM

File: CODE-OF-CONDUCT.md
Comment (catblade) [reply CODE-OF-CONDUCT.md:3]: This is still outstanding as an issue.

Comment (leonardpahlke) [reply CODE-OF-CONDUCT.md:3]: @catblade the file contains only a reference to the CNCF CoC (no more information)

**Review by catblade — APPROVED** [review]

**Pull request merged**
